test/provider/AppDirectory.unittest.ts

```diff
@@ -119,15 +146,15 @@ describe('AppDirectory Unit Tests', () => {
 
             it('Use cached applications when the URLs are the same', async () => {
```

Comment:
Is this test meaningfully different from the one below? Are we trying to test that `fetch` doesn't get called, because the cache can be use? Because fetch does get called in this test

Reply:
Yes they are both meaningful, but not in it's current written state.  In summary this tests description should be changed to something like `Fetches and updates directory when URLs are the same` and add some logic to change what is returned from the fetch.  The cache should only be used if the fetch fails which the second test covers.  

Speaking with @pjbroadbent, we are going to evaluate all the of the tests once the final story (Desktop owners file - SERVICE-516) is implemented as it will change a lot of the existing logic and we can do cleanup then.